### PR TITLE
8330852: All callers of JvmtiEnvBase::get_threadOop_and_JavaThread should pass current thread explicitly

### PR DESCRIPTION
**Pull request by alexmenkov**
Some cleanup related to JvmtiEnvBase::get_threadOop_and_JavaThread method

Testing: tier1-6

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8330852](https://bugs.openjdk.org/browse/JDK-8330852): All callers of JvmtiEnvBase::get_threadOop_and_JavaThread should pass current thread explicitly (**Enhancement** - P4)

### Reviewers
 * [Serguei Spitsyn](https://openjdk.org/census#sspitsyn) (@sspitsyn - **Reviewer**)
 * [Chris Plummer](https://openjdk.org/census#cjplummer) (@plummercj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18986/head:pull/18986` \
`$ git checkout pull/18986`

Update a local copy of the PR: \
`$ git checkout pull/18986` \
`$ git pull https://git.openjdk.org/jdk.git pull/18986/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18986`

View PR using the GUI difftool: \
`$ git pr show -t 18986`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18986.diff">https://git.openjdk.org/jdk/pull/18986.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18986#issuecomment-2080218439)

### CODE REVIEW

File: src/hotspot/share/prims/jvmtiEnvBase.cpp
Comment (plummercj) [inline src/hotspot/share/prims/jvmtiEnvBase.cpp:1976]: I think a good cleanup would be to also replace `current` with `current_thread`, although I'm not sure how common each are. I see 3 `current` references in this webrev.

Comment (alexmenkov) [reply src/hotspot/share/prims/jvmtiEnvBase.cpp:1976]: Looks like in JVMTI `current_thread` is more common (and `current` is usually used in runtime :)

**Review by sspitsyn — COMMENTED** [review]
This looks good in general. Thank you for taking care about it. The Runtime team suggested to use the name `current` for the current thread. The plan is to unify the JVMTI code as well. So, I'd suggest to rename all occurrences of the `current_thread` that you touched to the `current`. It will be a good step in a right direction.

File: src/hotspot/share/prims/jvmtiEnvBase.cpp
Comment (sspitsyn) [reply src/hotspot/share/prims/jvmtiEnvBase.cpp:1976]: The plan is to unify this with the approach used by the Runtime team.

Comment (alexmenkov) [reply src/hotspot/share/prims/jvmtiEnvBase.cpp:1976]: Replaced all touched "current_thread" and "calling_thread" with "current"

**Comment by plummercj** [comment]
Given the number of `current` renames (which distracts from the core change in this PR), and given that not all occurrences were renamed (only those that were touched), I think it would be best to leave the rename for a separate PR.

**Comment by alexmenkov** [comment]
ok, I'll revert last update and rename current_thread to current only in few places where new variable is introduced

**Review by sspitsyn — APPROVED** [review]
Looks good.

**Comment by alexmenkov** [comment]
Ping. Can I get second review please.

**Review by plummercj — APPROVED** [review]
Looks good.

**Pull request closed without merging**
